crates/api_common/src/post.rs

```diff
@@ -162,7 +162,7 @@ pub struct MarkPostAsRead {
 #[cfg_attr(feature = "full", ts(export))]
 /// Hide a post from list views
 pub struct HidePost {
-  pub post_ids: Vec<PostId>,
+  pub post_id: PostId,
```

Comment:
Why change this back to only a single item per call?

Reply:
We have the opposite request for jerboa: we want to have the result of that mark_read or hide_post, which isn't as clean when inserting multiple post ids. 

mlem and any other client could do these api requests individually also, and then they'd also have the benefit of receiving a result that says whether it succeeded for that item.

Reply:
Then how about separate endpoints for marking a single item as read and marking multiple as read?

Reply:
It just seems pointless to have that, considering that its easy to do separate requests for each one anyway, and you get back the thing changed when it succeeded. 

There are also a ton more operations that could take in multiple ids, (save, resolve, and probably a lot more), but its easy

Reply:
Hi,

Our use case for the batch mark read endpoint is to support mark read on scroll--we didn't want to build a "scroll fast to spam the instance" feature, and there's less overhead if we send these requests as batches rather than singletons.

Under normal circumstances, the highest rate we would feasibly generate is ~100 requests/second if a user is just scrolling as fast as loading will allow. If that's an acceptable rate from the instance side, then we don't have a hard technical blocker to migrating from batch to singletons.

>  Maybe hide post could return PostView in case of a single id, or success in case of multiple ids. But then its not type safe anymore.

Would returning an array of `PostView` be a feasible solution here? That would retain batching capabilities while also providing the richer response requested from Jerboa.

Reply:
The current rate limit for `MarkPostAsRead`, is the message rate limit, which by default is 180 / minute, and these aren't DB intensive operations, so it should be fine.

A better long-term solution, might be to add a back-end user setting for `auto_mark_posts_as_read`, and then everything that comes back from `ListPosts` (and not just `GetPost`), would automatically get marked. Then no additional API calls are required. 

It might not be *scroll exact*, but if the default fetch count isn't too large, then there's not much of a chance of getting too many unread posts marked as read.

Its probably not a good idea to try to return an array of postviews. DB-wise that would require making these db operations individual ones anyway.

Reply:
> A better long-term solution, might be to add a back-end user setting for auto_mark_posts_as_read, and then everything that comes back from ListPosts (and not just GetPost), would automatically get marked. Then no additional API calls are required.

@EricBAndrews how do you feel about this as an option? Then we do have to deal with any complicated batching, and its a setting available to all apps.

Reply:
> everything that comes back from ListPosts (and not just GetPost), would automatically get marked. 

This wouldn't work for apps that prefetch  ListPosts (Something i was planning to do at some point for Jerboa. This way you can prefetch the images and stop the feed from jumping when a image loads in)

Reply:
Yep, #4704 adds an image_details to the responses which has the dimensions.

Reply:
> Then we do have to deal with any complicated batching, and its a setting available to all apps.

I am strongly opposed to adding new user settings that change how API endpoints operate, unless it solves something that is impossible client-side.

It makes assumptions on the business logic of apps, which has bitten us in the past and confused users pretty badly (see: my arch nemesis: the "Show Read Posts" setting)